diff --git a/third_party/tsl/third_party/gpus/cuda/hermetic/cuda_redist_versions.bzl b/third_party/tsl/third_party/gpus/cuda/hermetic/cuda_redist_versions.bzl
index 71c1ee163e..46d1e761bb 100644
--- a/third_party/tsl/third_party/gpus/cuda/hermetic/cuda_redist_versions.bzl
+++ b/third_party/tsl/third_party/gpus/cuda/hermetic/cuda_redist_versions.bzl
@@ -156,6 +156,7 @@ REDIST_VERSIONS_TO_BUILD_TEMPLATES = {
             "555": "//third_party/gpus/cuda/hermetic:cuda_driver.BUILD.tpl",
             "550": "//third_party/gpus/cuda/hermetic:cuda_driver.BUILD.tpl",
             "545": "//third_party/gpus/cuda/hermetic:cuda_driver.BUILD.tpl",
+            "535": "//third_party/gpus/cuda/hermetic:cuda_driver.BUILD.tpl",
         },
     },
     "cuda_nccl": {
